addons/point_of_sale/static/src/js/models.js

```diff
@@ -127,13 +127,21 @@ exports.PosModel = Backbone.Model.extend({
                 progress: function(prog){
                     self.chrome.loading_progress(prog);
                 },
-            }).then(function(){
-                if(self.config.iface_scan_via_proxy){
-                    self.barcode_reader.connect_to_proxy();
-                }
-            }).always(function(){
-                done.resolve();
-            });
+            }).then(
+                function(){
+                        if(self.config.iface_scan_via_proxy){
+                            self.barcode_reader.connect_to_proxy();
+                        }
+                        done.resolve();
+                },
+                function(statusText, url){
+                        if (statusText == 'error' && window.location.protocol == 'https:') {
```

Comment:
is the "error", the only info we have? can't we make sure it is actually an error linked to the certificate and not something else (e.g. bad ip address)

Reply:
yes, a https error gives a empty reply. If the ip is wrong most likely it will not give a error but a timeout. So indeed, I must agree this is not perfect, but is seems to be the only option.